### ISSUE TITLE
Hang on invalid nest

### ISSUE THREAD

**Issue by max19** [urbit/urbit#833]
```
> =|  sut/span
  =+  ^=   q  |.((~(nest ut sut) | (span %noun)))
  =+  ^=  w  q(sut 0)
  (mock [w -:w] |=({* *} ~))
````

Bails at [ut_nest.c:198](https://github.com/urbit/urbit/blob/83d0f88710953d0674b0e8da15ae39721c218228/jets/f/ut_nest.c#L198), attempts to print a big trace with `u3m_p` and hangs. If `u3m_p` is bypassed, the worker process crashes and ends normally.

The reason this is relevant is that arvo's prettyprinter calls [something like this](https://github.com/urbit/arvo/blob/e10dc26beabcc08f67b88ecf3a98d17bd2d814b1/sys/hoon.hoon#L7995) for printing spans. If the value is not actually a span (for example it is testing a fork), this crash can happen.

**EDIT** (jtobin):

Still relevant as of 2019/06/19, except replace `span` with `type`.

**Comment by Fang-** [comment]
Confirmed to still be an issue.

**Comment by joemfb** [comment]
This should be reinvestigated once #1422 is merged

**Comment by jtobin** [comment]
It appears this has indeed been fixed (presumably by #1422):

```
[%2 p=~[[%leaf p="repo-fltt"]]]
```

Closing accordingly.

(cc @joemfb, @pilfer-pandex)
